### ISSUE TITLE
Refactor request zone arguments to be consistent

### ISSUE THREAD

**Issue by Temikus** [fog/fog-google#134]
Our request zone argument mappings are all over the place:

```
λ grep -Rh '"zone" => ' lib/fog/compute/google/requests | sort | uniq -c
   2                                    "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{@project}/zones/#{zone}",
  10             "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{@project}/global",
   6             "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{@project}/zones/#{zone_name}",
   1             "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{@project}/zones/#{zone}",
   1             "zone" => object["zone"],
  13             "zone" => zone
   2             "zone" => zone,
   1             "zone" => zone.split("/")[-1]
   1             "zone" => zone.split("/")[-1],
  14             "zone" => zone_name
   7             "zone" => zone_name,
   1             "zone" => zone_view.zone,
```

This will need to be refactored.

**Comment by Temikus** [comment]
@icco @plribeiro3000 What do you think should be the standard for the request input?
Self-link? Name? Object?

**Comment by icco** [comment]
I prefer name, sad it's slipped so much, I feel like we did a clean up like this a while back...

**Comment by plribeiro3000** [comment]
I'm afraid i don't have enough knowledge of GCE to have an opinion on that matter. 😅

**Comment by Temikus** [comment]
Something to think about additonally:
- Create an abstraction model for accepting `Server` model and other models as input in the requests, in addition to self_links. See #133 for context.
I'll think about abstracting this somehow.

**Comment by Temikus** [comment]
This was fixed in `1.0.0` transition, funnily enough.

```
λ grep -RH '"zone" => ' . | uniq
./lib/fog/compute/google/mock.rb:                      "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{@project}/zones/us-central1-a",
./lib/fog/compute/google/mock.rb:                      "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{key}/zones/us-central1-a",
./lib/fog/compute/google/mock.rb:                                          "zone" => zone,
./lib/fog/compute/google/mock.rb:                      "zone" => "https://www.googleapis.com/compute/#{api_version}/projects/#{key}/zones/us-central1-a",
```
